app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.onboarding.calculatorkotlin.mvp.model
+
+import com.onboarding.calculatorkotlin.mvp.CalculatorContract
+import com.onboarding.calculatorkotlin.util.Constants.ANY
+import com.onboarding.calculatorkotlin.util.Constants.EMPTY_STRING
+import com.onboarding.calculatorkotlin.util.Constants.MINUS
+import com.onboarding.calculatorkotlin.util.Constants.ONE_VALUE_REMOVED
+import com.onboarding.calculatorkotlin.util.Constants.TWO_CHARGED
+
+class CalculatorModel : CalculatorContract.Model {
+
+    private var firstOperand = EMPTY_STRING
+    private var secondOperand = EMPTY_STRING
+    private var operator = EMPTY_STRING
+
+    override fun setValue(value: String) {
+        if (operator.isEmpty()) {
+            firstOperand += value
+        } else {
+            secondOperand += value
+        }
+    }
+
+    override fun setOperator(operator: String) {
+        if (firstOperand.isNotEmpty()) {
+            this.operator = operator
+        }
+    }
+
+    private fun getCharged(): Int {
+        var charged = 0
+        if (firstOperand.isNotEmpty()) {
+            charged++
+            if (operator.isNotEmpty()) {
+                charged++
+                if (firstOperand.isEmpty()) {
+                    charged++
+                }
+            }
+        }
+        return charged
+    }
+
+    override fun addMinus() {
+        val charged = getCharged()
+        if (charged == ANY || charged == TWO_CHARGED) {
```

Comment:
I think getCharged is not necessary. Here you can said if (firstOpernd.isempty and operand.isEmpty )
setValue
else 
setOperand

Reply:
Solved in call